yaml.go

```diff
@@ -96,6 +96,61 @@ func JSONToYAML(j []byte) ([]byte, error) {
 	return yaml.Marshal(jsonObj)
 }
 
+// internalYamlUnmarshal works like yaml.Unmarshal but supports streams.
+func internalYamlUnmarshal(b []byte, out interface{}) error {
+	return internalYamlUnmarshalWrapper(false, b, out)
+}
+
+// internalYamlUnmarshalStrict works like yaml.UnmarshalStrict but supports streams.
+func internalYamlUnmarshalStrict(b []byte, out interface{}) error {
+	return internalYamlUnmarshalWrapper(true, b, out)
+}
+
+// internalYamlUnmarshalWrapper is a wrapper around yaml.Decoder that allows
+// for decoding yaml streams.
+func internalYamlUnmarshalWrapper(strict bool, b []byte, out interface{}) error {
+	outV := reflect.ValueOf(out)
+	if !outV.CanSet() {
+		outV = outV.Elem()
+	}
+
+	if len(bytes.TrimSpace(b)) == 0 {
+		outV.Set(reflect.Zero(outV.Type()))
+		return nil
+	}
+
+	decoder := yaml.NewDecoder(bytes.NewBuffer(b))
+	decoder.SetStrict(strict)
+
+	var outs []interface{}
+	for {
+		var tmp interface{}
+		err := decoder.Decode(&tmp)
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			return err
+		}
+		outs = append(outs, tmp)
+	}
+
+	if len(outs) == 0 {
```

Comment:
this branch is not needed as a no-op.

Reply:
i will leave this question to the api-machinery folks, but this makes sense to me:
> I think you would expect null, so I think I need to set out to zero in this branch.